### PR TITLE
replace placeholder with fetch; outsource URL into config.js; fix layout

### PR DESCRIPTION
**Pull request by makitango**

### CODE REVIEW

File: components/card/card.js [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
+        <dd class="card__info-description">${props.type}</dd>
         <dt class="card__info-title">Occurrences</dt>
-        <dd class="card__info-description">${characterOccurrences}</dd>
+        <dd class="card__info-description">${props.episode.length}</dd>
       </dl>
-    </div>`
+    </div>`;
   const newCard = document.createElement("li");
   newCard.classList.add("card");
   newCard.innerHTML = cardContent;
```
Comment (ahohnsen) [inline components/card/card.js]: You can also just paste the card content directly in here, without declaring a variable for it.
```suggestion
  newCard.innerHTML =  
  `<div class="card__image-container">
      <img
        class="card__image"
        src=${props.image}
        alt="${props.name}"
      />
      <div class="card__image-gradient"></div>
    </div>
    <div class="card__content">
      <h2 class="card__title">${props.name}</h2>
      <dl class="card__info">
        <dt class="card__info-title">Status</dt>
        <dd class="card__info-description">${props.status}</dd>
        <dt class="card__info-title">Gender</dt>
        <dd class="card__info-description">${props.gender}</dd>
        <dt class="card__info-title">Species</dt>
        <dd class="card__info-description">${props.species}</dd>
        <dt class="card__info-title">Type</dt>
        <dd class="card__info-description">${props.type}</dd>
        <dt class="card__info-title">Occurrences</dt>
        <dd class="card__info-description">${props.episode.length}</dd>
      </dl>
    </div>`;
```

File: config.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const ramCharacters = "https://rickandmortyapi.com/api/character";
```
Comment (ahohnsen) [inline config.js:1]: As you probably won't have anything else inside this config file, I suggest to just enter the url in the `index.js` file directly. Of course, it also works this way but to me it is a bit much to have a separate file for only one url.

File: index.js [hunk 3]
```diff
@@ -12,3 +15,24 @@ const pagination = document.querySelector('[data-js="pagination"]');
...
+        cardContainer.append(card);
+      });
+      return data;
+    } else {
+      console.error("response not OK");
+    }
+  } catch (error) {
+    console.error("API not available: ", error);
+  }
+}
```
Comment (ahohnsen) [inline index.js]: I like that you use a try/catch block here 🙂 

In this case, you directly create the character cards from the data. You do not need to return the data, as you won't do anything else with it.
```suggestion
async function fetchCharacters() {
  try {
    const response = await fetch(ramCharacters);
    if (response.ok) {
      const data = await response.json();
      const characterData = data.results;
      cardContainer.innerHTML = "";
      characterData.forEach((character) => {
        const card = createCharacterCard(character);
        cardContainer.append(card);
      });
    } else {
      console.error("response not OK");
    }
  } catch (error) {
    console.error("API not available: ", error);
  }
}
```

**Review by ahohnsen — CHANGES_REQUESTED** [review]
Well done 🙂 

Just remove the `return` from your fetch function. I also added some other minor suggestions but of course, you can decide for yourself.

File: config.js (hunk 2 above)
Comment (makitango) [reply config.js:1]: We just executed the will of his majesty, Aljoscha, who wanted the URL outside of his realm.

File: components/card/card.js (hunk 1 above)
Comment (makitango) [reply components/card/card.js]: We deleted the redundant variable ✅

**Pull request merged**